File: python/algorithm_problem/programmers/ServerScaleOutCount.py
def answer(players, m, k):
    server_add = [p // m for p in players]

    for i in range(len(server_add)):
        current = server_add[i]
        if current > 0:
            limit = min(i + k, len(server_add))
            for j in range(i + 1, limit):
                updated_value = server_add[j] - current
                server_add[j] = max(updated_value, 0)

    return sum(server_add)

if __name__ == "__main__":
    players = [0, 2, 3, 3, 1, 2, 0, 0, 0, 0, 4, 2, 0, 6, 0, 4, 2, 13, 3, 5, 10, 0, 1, 5]
    m = 3
    k = 5

    print(answer(players, m, k))

File: kotlin/src/algorithm-problem/programmers/ServerScaleOutCount.kt
package `algorithm-problem`.programmers


fun main() {
    val players = intArrayOf(0, 2, 3, 3, 1, 2, 0, 0, 0, 0, 4, 2, 0, 6, 0, 4, 2, 13, 3, 5, 10, 0, 1, 5)
    val m = 3
    val k = 5

    println(answer(players, m, k))
}

fun answer(players: IntArray, m: Int, k: Int): Int {
    val serverAdd = players.map { it / m }.toMutableList()

    for (i in serverAdd.indices) {
        val current = serverAdd[i]
        if (current > 0) {
            val limit = minOf(i + k, serverAdd.size)
            for (j in i + 1 until limit) {
                val updatedValue = serverAdd[j] - current
                serverAdd[j] = maxOf(updatedValue, 0)
            }
        }
    }

    return serverAdd.sum()
}